modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/utils/imagePropsTransformer.ts

```diff
@@ -3,21 +3,82 @@
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
 
-export default function imagePropsTransformer(
-	imageData: any | string
-): React.ImgHTMLAttributes<HTMLImageElement> | string | undefined {
-	if (typeof imageData === 'string') {
-		return imageData;
-	}
+import getValueFromItem from '../../utils/getValueFromItem';
+
+interface IDocsAndMediaImageProps {
+	id: number;
+	link: {
+		href: string;
+		label: string;
+	};
+	name: string;
+}
 
-	if (typeof imageData === 'object') {
+interface IImageProps {
+	itemData?: any;
+	name?: string;
+	options?: {
+		label?: string;
+		labelKey?: string;
+		shape?: 'circle' | 'rounded' | 'user-icon';
+		size?: 'lg' | 'sm' | 'xl';
+	};
+	value?:
+		| string
+		| IDocsAndMediaImageProps
+		| React.ImgHTMLAttributes<HTMLImageElement>;
+}
 
-		// Docs & media object
+export default function imagePropsTransformer(
+	imageData: IDocsAndMediaImageProps | IImageProps | string | undefined
+): React.ImgHTMLAttributes<HTMLImageElement> | undefined {
+	let imageProps: React.ImgHTMLAttributes<HTMLImageElement> = {
+		alt: '',
+		src: undefined,
+	};
 
-		return {
-			alt: imageData?.link.label,
-			id: imageData?.id,
-			src: imageData?.link.href,
+	if (!imageData || !Object.keys(imageData).length) {
+		return undefined;
+	}
+
+	if (typeof imageData === 'string') {
+		imageProps = {...imageProps, src: imageData};
+	}
+	else if ('value' in imageData && typeof imageData.value === 'string') {
+		imageProps = {
+			...imageProps,
+			alt:
+				imageData.options?.label ||
+				(imageData.options?.labelKey
+					? getValueFromItem(
+							imageData.itemData,
+							imageData.options.labelKey
+					  )
+					: ''),
+			src: imageData.value,
 		};
 	}
+	else if ('value' in imageData && typeof imageData.value === 'object') {
+		imageProps =
+			'link' in imageData.value
+				? {
+						...imageProps,
+						alt: imageData.value.link.label,
+						src: imageData.value.link.href,
+				  }
+				: {
+						...imageProps,
+						alt: imageData.value.alt,
+						src: imageData.value.src,
+				  };
+	}
+	else if ('link' in imageData) {
+		imageProps = {
```

Comment:
Need to confirm if Documents and Media allows providing a valid `alt` content. `label` and `name` properties return always the name of the image with the extension. a11y issue.